includes/distributor-customizations/global.php

```diff
@@ -1,23 +1,81 @@
 <?php
 /**
- * This files contains the global tweaks that are loaded by default in every site in the Network (bot Hub and Nodes)
+ * This files contains the global tweaks that are loaded by default in every site in the Network (both Hub and Nodes).
  *
  * @package newspack-network
  */
 
 /**
- * =========================================
- * ======== Post publication date ==========
- * =========================================
- * Keep the publication date on the new pushed post.
+ * Send primary category slug to the Node along with post update data.
  *
- * This filter is used to filter the arguments sent to the remote server during a push. The below code snippet passes the original published date to the new pushed post and sets the same published date instead of setting it as per the current time.
+ * @param WP_Post $post The post object.
+ * @param bool    $is_pulling Whether the post is being pulled from the remote site.
+ */
+function newspack_network_get_primary_category_slug( $post, $is_pulling = false ) {
+	if ( $is_pulling ) {
+		// When pulling content, the post will be the remote site post (not on the WP instance that executes this code).
+		// The category slug has to be read from the data on the post object.
+		if ( ! isset( $post->meta['_yoast_wpseo_primary_category'] ) ) {
+			return;
+		}
+		$primary_category_id = reset( $post->meta['_yoast_wpseo_primary_category'] );
+		if ( ! $primary_category_id ) {
+			return;
+		}
+		$maybe_primary_categories = array_filter(
+			$post->terms['category'],
+			function( $category ) use ( $primary_category_id ) {
+				return (int) $category['term_id'] === (int) $primary_category_id;
+			}
+		);
+		$maybe_primary_category   = reset( $maybe_primary_categories );
+		if ( $maybe_primary_category ) {
+			return $maybe_primary_category['slug'];
+		}
+	} elseif ( class_exists( 'WPSEO_Primary_Term' ) ) {
+		// When pushing, the post will be the post on this site.
+		// The category exists on the site which executes this code, so it can be retrieved via Yoast.
+		$primary_term = new WPSEO_Primary_Term( 'category', $post->ID );
+		$category_id  = $primary_term->get_primary_term();
+		if ( $category_id ) {
+			$category = get_term( $category_id );
+			return $category->slug;
+		}
+	}
+}
+
+/**
+ * Fix primary category on the Node.
+ *
+ * @param WP_Post $post The post object.
+ */
+function newspack_network_fix_primary_category( $post ) {
```

Comment:
nit: remove the "hub" jargon from this method

Reply:
Done in 29ee85fa914a5c0e8688d28787fbed8dbebdee0d